source/common/protobuf/utility.h

```diff
@@ -310,6 +310,14 @@ class DurationUtil {
    * @throw OutOfRangeException when duration is out-of-range.
    */
   static uint64_t durationToSeconds(const ProtobufWkt::Duration& duration);
+
+  /**
+   * Writes a time_point<system_clock> (SystemTime) to a protobuf Timestamp, by way of time_t.
```

Comment:
Also, do you want to go via `time_t` or some fixed granularity such as milliseconds?